### PR TITLE
checkpoint: make run func generic

### PR DESCRIPTION
**Pull request by xiang90**
Make run func generic. Preparing for adding etcd checkpoint support.

### CODE REVIEW

**Comment by xiang90** [comment]
If this approach looks good to you, I will continue to clean up the manifest file part.

File: cmd/checkpoint/main.go [hunk 1]
```diff
@@ -30,8 +30,9 @@ const (
 )
 
 var (
-	tempAPIServer      = []byte("temp-apiserver")
-	kubeAPIServer      = []byte("kube-apiserver")
+	tempAPIServer = "temp-apiserver"
```
Comment (derekparker) [inline cmd/checkpoint/main.go]: Since these are just strings now instead of byte slices, make then `const`.

File: cmd/checkpoint/main.go [hunk 2]
```diff
@@ -115,36 +116,35 @@ func getPodsFromKubeletAPI() []byte {
...
+		tempPodSeen = tempPodSeen || isPod(p, tn, ns)
+		if actualPodSeen && tempPodSeen {
 			return true
 		}
 	}
 	return false
 }
 
-func kubeSystemAPIServerRunning(pods v1.PodList, client clientset.Interface) bool {
+func actualPodRunning(pods v1.PodList, client clientset.Interface, n, ns string) bool {
```
Comment (derekparker) [inline cmd/checkpoint/main.go]: nit: Maybe would be better named generically `isPodRunning` or just `podRunning`?

Comment (xiang90) [inline cmd/checkpoint/main.go]: will change to is podRunning.

**Comment by xiang90** [comment]
@derekparker I will continue to make the manifest part more generic.

**Comment by xiang90** [comment]
@derekparker @aaronlevy 

All fixed. And I changed manifest part to be generic too. PTAL. I do not have setup to test checkpointer yet. It would be great, if you can help me to do a quick test. Thank you in advance.

**Comment by aaronlevy** [comment]
@pbx0 this would be a good candidate for your destructive tests. Testing this will be a little wonky though - we would likely need to build a checkpoint image from this commit, then use that image in the checkpoint manifest in this PR.

**Comment by pbx0** [comment]
@aaronlevy: LGTM I tested it against the basic destructive test here: https://github.com/kubernetes-incubator/bootkube/pull/201

**Pull request merged**